lib/web_ui/test/engine/window_test.dart

```diff
@@ -407,6 +407,145 @@ Future<void> testMain() async {
+    expect(lockCalls, <String>[]);
+    expect(unlockCount, 0);
+    await domWindow.screen!.orientation!.lock('hi');
+    domWindow.screen!.orientation!.unlock();
+    expect(lockCalls, <String>['hi']);
+    expect(unlockCount, 1);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>['DeviceOrientation.portraitUp']), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>['DeviceOrientation.portraitUp']);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>['DeviceOrientation.portraitDown']), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>['DeviceOrientation.portraitDown']);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>['DeviceOrientation.landscapeLeft']), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>['DeviceOrientation.landscapeLeft']);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>['DeviceOrientation.landscapeRight']), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>['DeviceOrientation.landscapeRight']);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>[
+      'DeviceOrientation.portraitUp',
+      'DeviceOrientation.landscapeRight'
+    ]), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>[
+      'DeviceOrientation.portraitUp',
+      'DeviceOrientation.landscapeRight'
+    ]);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>[
+      'DeviceOrientation.portraitUp',
+      'DeviceOrientation.portraitDown',
+      'DeviceOrientation.landscapeRight'
+    ]), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>[
+      'DeviceOrientation.portraitUp',
+      'DeviceOrientation.portraitDown',
+      'DeviceOrientation.landscapeRight'
+    ]);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>[
+      'DeviceOrientation.portraitUp',
+      'DeviceOrientation.portraitDown',
+      'DeviceOrientation.landscapeLeft',
+      'DeviceOrientation.landscapeRight'
+    ]), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>[
+      'DeviceOrientation.portraitUp',
+      'DeviceOrientation.portraitDown',
+      'DeviceOrientation.landscapeLeft',
+      'DeviceOrientation.landscapeRight'
+    ]);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    expect(await sendSetPreferredOrientations(<dynamic>[]), isTrue);
+    expect(await sendGetPreferredOrientations(), <dynamic>[]);
+    lockCalls.clear();
+    unlockCount = 0;
+
+    simulateError = true;
+    expect(await sendSetPreferredOrientations(<dynamic>['DeviceOrientation.portraitDown']), isFalse);
+    expect(await sendGetPreferredOrientations(), <dynamic>['DeviceOrientation.portraitDown']);
```

Comment:
Is this the desired outcome if we failed to lock onto `portraitDown`?

Reply:
Thank you for the comments! I will fix those three above in the following commit.

Yes, I believe `GetPreferredOrientations` should return requested orientations without any changes. Please, consider the following case:
```dart
// somewhere in my app
final requested = [DeviceOrientation.portraitDown];
await SystemChrome.setPreferredOrientations(requested);

// later in a third party package
final preferredByMainApp = await SystemChrome.getPreferredOrientations();
final requestedByPackage = [DeviceOrientation.portraitUp];
await SystemChrome.setPreferredOrientations(requestedByPackage);

// the package can restore preferred orientations
await SystemChrome.setPreferredOrientations(preferredByMainApp);
```
I thought, if `getPreferredOrientations` will return an empty list, than the last call to `setPreferredOrientations` wouldn't be the same as the first one - it seems wrong to me.